src/cpp/src/sequence_group.hpp

```diff
@@ -126,12 +126,26 @@ class Sequence {
     // Each KV block can be uniquely identified by 
     // the tokens within the block and the tokens in the prefix before the block.
     // hash(prefix tokens + block tokens) <--> KV Block
-    size_t get_hash(size_t content_length, const ov::genai::TokenIds& prompt_ids) const {
-        std::vector<int64_t> content;
+    size_t get_hash(const std::vector<size_t>& prefix_hashes, size_t block_size, size_t content_length, const ov::genai::TokenIds& prompt_ids) const {
```

Comment:
do you think we can store `prefix_hashes` directly within sequence? to avoid passing this information to this method each time.

Also, we can pass weak_ptr<SequenceGroup> to its Sequences during sequence creation. So, prompt ids and context-length can be taken by sequence by calling sequence group methods. 

Ideally, I think get_hash should have no arguments and return hashes based on current context length. Do you think it's reasonable?

Reply:
> do you think we can store `prefix_hashes` directly within sequence?
Yes, done.

> Also, we can pass weak_ptr to its Sequences during sequence creation.
Done.

> Ideally, I think get_hash should have no arguments and return hashes based on current context length.
There are still 2 cases when we need to pass `content_length` to `get_hash()`. 
1. When we restore blocks from `cached_blocks`. We need to calculate hashes block by block, not only final block for full content.
2. When we allocate more than 1 block at prompt phase. We need to set hashes for all allocated blocks, so we need intermediate hashes. 

So I made `content_length` optional. If not set, `content_length` is taken from `sequence_group`.